isis/tests/FunctionalTestsJigsaw.cpp

```diff
@@ -0,0 +1,246 @@
+
+  // Check for the correct header output format and csv file structure for the residuals.csv file
+  header = CSVReader(residualsOutput,
+                     false, 0, ',', false, true);
+
+  numColumns = header.columns();
+  numRows = header.rows();
+
+  ASSERT_EQ(numColumns, 7);
+  ASSERT_EQ(numRows, 972);
+
+  // Validate the header information is correct
+  csvLine = header.getRow(0);
+  compareCsvLine(csvLine, "x image,y image,Measured,Measured,sample,line,Residual Vector");
+  csvLine = header.getRow(1);
+  compareCsvLine(csvLine, "Point,Image,Image,coordinate,coordinate,Sample,Line,residual,residual,Magnitude");
+  csvLine = header.getRow(2);
+  compareCsvLine(csvLine, "Label,Filename,Serial Number,(mm),(mm),(pixels),(pixels),(pixels),(pixels),(pixels),Rejected");
+
+  // Check line/sample and residuals 
+  // Check all measures
+  ControlMeasure* measure;
+  for (int i=3; i < numRows; i++) {
+    csvLine = header.getRow(i);
+    EXPECT_NO_THROW({
+      outputPoint = outputNet.GetPoint(csvLine[0]);
+    });
+    EXPECT_NO_THROW({
+      measure = outputPoint->GetMeasure(csvLine[2]);
+   });
+    // Compare sample, line, residuals
+    EXPECT_NEAR(csvLine[5].toDouble(), measure->GetSample(), 0.000001);
+    EXPECT_NEAR(csvLine[6].toDouble(), measure->GetLine(), 0.000001);
+    EXPECT_NEAR(csvLine[7].toDouble(), measure->GetSampleResidual(), 0.000001);
+    EXPECT_NEAR(csvLine[8].toDouble(), measure->GetLineResidual(), 0.000001);
+    EXPECT_NEAR(csvLine[9].toDouble(), measure->GetResidualMagnitude(), 0.000001);
+  }
+
+  // Spot check a few measures for hard-coded values:
+  compareCsvLine(header.getRow(14), "AS15_000031448,/tmp/qt_temp-l7wTTZ/cube1.cub,APOLLO15/METRIC/1971-07-31T14:00:53.547,    -24.91466687,     -8.24555718,   4109.77150653,   2450.19288272,-0.00343036,      0.70304341,      0.70305178", 2); 
+
+  compareCsvLine(header.getRow(142), "AS15_000032200,/tmp/qt_temp-l7wTTZ/cube2.cub,APOLLO15/METRIC/1971-07-31T14:01:16.947,    -25.59176645,    -10.57595225,   4143.71597937,   2333.56318790,     -0.00372340,      0.48459237,      0.48460667", 2);
+  compareCsvLine(header.getRow(424), "AS15_000055094,/tmp/qt_temp-l7wTTZ/cube1.cub,APOLLO15/METRIC/1971-07-31T14:00:53.547,     20.35945982,     34.23830188,   1844.18431849,   4576.36730130,      0.00691810,     -0.57578795,      0.57582951", 2);
+  compareCsvLine(header.getRow(970), "AS15_test01,/tmp/qt_temp-l7wTTZ/cube3.cub,APOLLO15/METRIC/1971-07-31T14:01:40.346,     -5.04180936,    -34.53366079,   3115.51026031,   1134.42313078,     -3.07166949,      1.44947401,      3.39648765", 2);
+
+  // Test output network size
+  ControlNet inputNet("$ISISROOT/../isis/tests/data/apolloNetwork/apollo.net");
```

Comment:
I recall you having issues with the test data area but this should probably be a relative path or a path in fixture (as Jenkins doesn't seem to complain with relative paths).

Reply:
Good catch, didn't realize I'd left this in.